src/mediadev.c

```diff
@@ -0,0 +1,84 @@
+/**
+ * @file mediadev.c  Media device
+ *
+ * Copyright (C) 2010 - 2018 Creytiv.com
+ */
+#include <re.h>
+#include <baresip.h>
+
+
+static void destructor(void *data)
+{
+	struct mediadev *dev = data;
+
+	mem_deref(dev->name);
+}
+
+
+int mediadev_add(struct list *dev_list, const char *name)
+{
+	struct mediadev *dev;
+	int err;
+
+	if (!dev_list || !str_isset(name))
+		return EINVAL;
+
+	dev = mem_zalloc(sizeof(*dev), destructor);
+	if (!dev)
+		return ENOMEM;
+
+	err = str_dup(&dev->name, name);
+	if (err)
+		goto out;
+
+	list_append(dev_list, &dev->le, dev);
+
+ out:
+	if (err)
+		mem_deref(dev);
+
+	return err;
+}
+
+
+struct mediadev *mediadev_find(const struct list *dev_list, const char *name)
+{
+	struct le *le;
+
+	for (le = list_head(dev_list); le; le = le->next) {
+
+		struct mediadev *dev = le->data;
+
+		if (!str_casecmp(dev->name, name))
+			return dev;
+	}
+
+	return NULL;
+}
+
+
+struct mediadev *mediadev_get_default(const struct list *dev_list)
+{
+	return (struct mediadev*)list_head(dev_list)->data;
+}
```

Comment:
if the list is empty, this function will crash.

the function should return NULL if the list is NULL or empty.

also, the casting is not necessary.